File: python/super_resolution_aksk_demo.py
# -*- coding:utf-8 -*-
from ais_sdk.utils import encode_to_base64
from ais_sdk.utils import decode_to_wave_file
from ais_sdk.super_resolution import super_resolution_aksk
from ais_sdk.utils import init_global_env
import json

if __name__ == '__main__':
    # Region currently supports Beijing(cn-north-1) and Hong Kong(ap-southeast-1)
    init_global_env('cn-north-1')

    #
    # access image super resolution enhance,post data by sk,sk
    #
    app_key = '*************'
    app_secret = '************'

    # call interface use base64 file
    result = super_resolution_aksk(app_key, app_secret, encode_to_base64('data/super-resolution-demo.png'), 3)
    result_obj = json.loads(result)
    decode_to_wave_file(result_obj['result'], 'data/super-resolution-demo-aksk.png')

File: javascript/super_resolution_aksk_demo.js
/**
 * 超分重建服务ak,sk方式请求的使用示例
 */
var supresol = require("./ais_sdk/super_resolution");
var utils = require("./ais_sdk/utils");

// 初始化服务的区域信息，目前支持北京1(cn-north-1)、香港(ap-southeast-1)等区域信息
utils.initRegion(region="cn-north-1");

var app_key = "*************";
var app_secret = "************";

var filepath = "./data/super-resolution-demo.png";
var data = utils.changeFileToBase64(filepath);

supresol.super_resolution_aksk(app_key, app_secret, data, 3, "ESPCN", function (result) {
    var resultObj = JSON.parse(result);
    utils.getFileByBase64Str("./data/super-resolution-demo-aksk.png", resultObj.result);
});

